kernel/primred.mli

```diff
@@ -60,20 +61,25 @@ module type RedNativeEntries =
     val mkNInf : env -> elem
     val mkNaN : env -> elem
     val mkArray : env -> uinstance -> elem Parray.t -> elem -> elem
+
+    val eval_lazy : lazy_info -> elem -> elem
+    val mkApp : elem -> elem array -> elem
```

Comment:
If primitives should reduce to "values", then it probably makes sense to expose the current evaluation strategy as an argument. Something like `eval_app : elem -> elem array -> elem`.